### PR TITLE
refac(purify): Refactoring Purify

### PR DESCRIPTION
**Pull request by ssi02014**
## Summary

Hello 👋 @cure53 
I've done some refactoring around "purify.js".
Please review this pull request to see if it's appropriate.

After the modifications, the tests are all passing fine.🙏

## Tasks

- Refactoring purify.js

### CODE REVIEW

File: src/purify.js [hunk 1]
```diff
@@ -128,7 +130,7 @@ function createDOMPurify(window = getGlobal()) {
     }
   }
 
-  let trustedTypesPolicy;
+  let trustedTypesPolicy = null;
```
Comment (ssi02014) [inline src/purify.js:133]: Why not explicitly express that it has no value by initializing it to "null"?

File: src/purify.js [hunk 2]
```diff
@@ -393,16 +395,11 @@ function createDOMPurify(window = getGlobal()) {
...
-  const _parseConfig = function (cfg) {
+  const _parseConfig = function (cfg = {}) {
     if (CONFIG && CONFIG === cfg) {
       return;
     }
 
-    /* Shield configuration object from tampering */
-    if (!cfg || typeof cfg !== 'object') {
-      cfg = {};
-    }
```
Comment (ssi02014) [inline src/purify.js:404]: I think it could be removed by using `default parameters` for `cfg`.

File: src/purify.js [hunk 3]
```diff
@@ -831,8 +833,8 @@ function createDOMPurify(window = getGlobal()) {
    */
   const _initDocument = function (dirty) {
     /* Create a HTML document */
-    let doc;
-    let leadingWhitespace;
+    let doc = null;
+    let leadingWhitespace = null;
```
Comment (ssi02014) [inline src/purify.js:837]: Why not explicitly express that it has no value by initializing it to "null"?

**Comment by cure53** [comment]
Those refactoring look great, thank you :)

**Pull request merged**